src/main/scala/scorex/crypto/authds/avltree/batch/serialization/ProxyInternalNode.scala

```diff
@@ -12,11 +16,19 @@ class ProxyInternalNode[D <: Digest](protected var pk: ADKey,
                                     (implicit val phf: CryptographicHash[D])
   extends InternalProverNode(k = pk, l = null, r = null, b = pb)(phf) {
 
-  override def computeLabel: D = hf.prefixedHash(1: Byte, Array(b), leftLabel, rightLabel)
+  override def computeLabel: D = {
+    hf.hash(Array(InternalNodePrefix, b), leftLabel, rightLabel)
+  }
 
-  override def label: D = if (isEmpty) selfLabelOpt.getOrElse(computeLabel) else super.label
+  override def label: D = if (isEmpty) {
+    val l = selfLabelOpt.getOrElse(computeLabel)
+    labelOpt = Some(l)
```

Comment:
I think this line requires explanation. What this means? why it is necessary?

Reply:
addressed